assertions/assertions.go

```diff
@@ -1174,6 +1174,120 @@ func ShouldTimeEqual(actual interface{}, expected ...interface{}) error {
 	return fmt.Errorf("expected '%v' to be time equals to '%v' ", actualTime, expectedTime)
 }
 
+// ShouldJSONEqual receives exactly 2 JSON arguments and does a JSON equality check.
+// This means the keys can be in different order, and whitespace (except in keys or values) is ignored.
+// JSON can be an object or array (i.e. start with `{` or `[`).
+//
+// Example of testsuite file:
+//
+//	name: test ShouldJSONEqual
+//	vars:
+//	  json_expected: '{"a":1,"b":"foo"}'
+//	testcases:
+//	- name: test assertion
+//	  steps:
+//	  - type: exec
+//	    script: "echo '{ \"b\" : \"foo\", \"a\" : 1 }'"
+//	    assertions:
+//	      - result.systemout ShouldJSONEqual "{{.json_expected}}"
+func ShouldJSONEqual(actual interface{}, expected ...interface{}) error {
+	if err := need(1, expected); err != nil {
+		return err
+	}
+
+	expectedString, err := cast.ToStringE(expected[0])
+	if err != nil {
+		return err
+	}
+
+	// `actual` can be a map, slice or the string representation of a JSON object or array.
+	var actualMap map[string]interface{}
+	var actualSlice []interface{}
+	var obj bool
+	switch actual.(type) {
+	case map[string]interface{}:
```

Comment:
I think that the parser to detect the json could be more json compliant. There is a check for `{` or `[`, but a simple text or number is also a valid number, see all values on  https://www.json.org/json-en.html

Reply:
I prefer the first proposal: let the choice to the user and having the ShouldJSONEqual supporting the different types. Maybe not the easiest to implement, but definitely the easiest for users to understand if we say that ShouldJSONEqual supports the full JSON spec.

Reply:
Makes sense 👍 

Done in 5c05103a6d3b69b66ae3cde5e374f4f7a5e57a05

⚠️ JSON `null` are tricky, because Venom passes it as empty string and not Go's `nil`. Haven't checked if this is a fundamental limitation in the upstream callers in Venom, or if it can be changed. Do you know maybe?

And added to README in ba3769c72a0ba06a0a0daaaaf8fd663c27fe473f

Looking at the assertion list in the README it's a bit messy, I think it would make sense to sort the assertions alphabetically. But then the `ShouldJSONEqual` feels misplaced. => WDYT about naming it `ShouldEqualJSON` like originally proposed in #633, and then for consistency create `ShouldEqualTime` as an alias for `ShouldTimeEqual`, which is required for backward compatibility. The latter could be marked as deprecated in the README.

Reply:
About naming, `ShouldEqualJSON ` is ok too, perhaps we can add some alias in another pr than here if needed. Thank you for adding diffrents json values.
LGTM